---------------------------------------------
SLURM job ID:        16040
SLURM job node list: babbage
DATE:                Fri Jun  6 09:52:32 AM CEST 2025
---------------------------------------------
no change     /opt/spack/opt/spack/linux-rocky9-x86_64/gcc-13.2.0/miniconda3-22.11.1-tn534fvb4uy4wrf7m2zcwpiycdzlebd6/condabin/conda
no change     /opt/spack/opt/spack/linux-rocky9-x86_64/gcc-13.2.0/miniconda3-22.11.1-tn534fvb4uy4wrf7m2zcwpiycdzlebd6/bin/conda
no change     /opt/spack/opt/spack/linux-rocky9-x86_64/gcc-13.2.0/miniconda3-22.11.1-tn534fvb4uy4wrf7m2zcwpiycdzlebd6/bin/conda-env
no change     /opt/spack/opt/spack/linux-rocky9-x86_64/gcc-13.2.0/miniconda3-22.11.1-tn534fvb4uy4wrf7m2zcwpiycdzlebd6/bin/activate
no change     /opt/spack/opt/spack/linux-rocky9-x86_64/gcc-13.2.0/miniconda3-22.11.1-tn534fvb4uy4wrf7m2zcwpiycdzlebd6/bin/deactivate
no change     /opt/spack/opt/spack/linux-rocky9-x86_64/gcc-13.2.0/miniconda3-22.11.1-tn534fvb4uy4wrf7m2zcwpiycdzlebd6/etc/profile.d/conda.sh
no change     /opt/spack/opt/spack/linux-rocky9-x86_64/gcc-13.2.0/miniconda3-22.11.1-tn534fvb4uy4wrf7m2zcwpiycdzlebd6/etc/fish/conf.d/conda.fish
no change     /opt/spack/opt/spack/linux-rocky9-x86_64/gcc-13.2.0/miniconda3-22.11.1-tn534fvb4uy4wrf7m2zcwpiycdzlebd6/shell/condabin/Conda.psm1
no change     /opt/spack/opt/spack/linux-rocky9-x86_64/gcc-13.2.0/miniconda3-22.11.1-tn534fvb4uy4wrf7m2zcwpiycdzlebd6/shell/condabin/conda-hook.ps1
no change     /opt/spack/opt/spack/linux-rocky9-x86_64/gcc-13.2.0/miniconda3-22.11.1-tn534fvb4uy4wrf7m2zcwpiycdzlebd6/lib/python3.10/site-packages/xontrib/conda.xsh
no change     /opt/spack/opt/spack/linux-rocky9-x86_64/gcc-13.2.0/miniconda3-22.11.1-tn534fvb4uy4wrf7m2zcwpiycdzlebd6/etc/profile.d/conda.csh
no change     /u/scandussio/.bashrc
No action taken.
The following values were not passed to `accelerate launch` and had defaults used instead:
	`--num_processes` was set to a value of `1`
	`--num_machines` was set to a value of `1`
	`--mixed_precision` was set to a value of `'no'`
	`--dynamo_backend` was set to a value of `'no'`
To avoid this warning pass in values for each of the problematic parameters or run `accelerate config`.
ðŸ¦¥ Unsloth: Will patch your computer to enable 2x faster free finetuning.
ðŸ¦¥ Unsloth Zoo will now patch everything to make training faster!
INFO 06-06 09:53:09 [__init__.py:243] Automatically detected platform cuda.
wandb: Appending key for api.wandb.ai to your netrc file: /u/scandussio/.netrc
wandb: Currently logged in as: saracandussio (saracandussio-universit-degli-studi-di-trieste) to https://api.wandb.ai. Use `wandb login --relogin` to force relogin
==((====))==  Unsloth 2025.5.9: Fast Mistral patching. Transformers: 4.52.4. vLLM: 0.9.0.1.
   \\   /|    NVIDIA A100-PCIE-40GB. Num GPUs = 1. Max memory: 39.381 GB. Platform: Linux.
O^O/ \_/ \    Torch: 2.7.0+cu126. CUDA: 8.0. CUDA Toolkit: 12.6. Triton: 3.3.0
\        /    Bfloat16 = TRUE. FA [Xformers = 0.0.30. FA2 = False]
 "-____-"     Free license: http://github.com/unslothai/unsloth
Unsloth: Fast downloading is enabled - ignore downloading bars which are red colored!
Unsloth: vLLM loading unsloth/Phi-3-mini-4k-instruct-v0-bnb-4bit with actual GPU utilization = 49.43%
Unsloth: Your GPU has CUDA compute capability 8.0 with VRAM = 39.38 GB.
Unsloth: Using conservativeness = 1.0. Chunked prefill tokens = 1248. Num Sequences = 288.
Unsloth: vLLM's KV Cache can use up to 16.77 GB. Also swap space = 6 GB.
INFO 06-06 09:53:28 [__init__.py:31] Available plugins for group vllm.general_plugins:
INFO 06-06 09:53:28 [__init__.py:33] - lora_filesystem_resolver -> vllm.plugins.lora_resolvers.filesystem_resolver:register_filesystem_resolver
INFO 06-06 09:53:28 [__init__.py:36] All plugins in this group will be loaded. Set `VLLM_PLUGINS` to control which plugins to load.
INFO 06-06 09:53:46 [config.py:793] This model supports multiple tasks: {'reward', 'embed', 'generate', 'classify', 'score'}. Defaulting to 'generate'.
INFO 06-06 09:53:47 [config.py:2118] Chunked prefill is enabled with max_num_batched_tokens=1248.
Unsloth: vLLM Bitsandbytes config using kwargs = {'load_in_8bit': False, 'load_in_4bit': True, 'bnb_4bit_compute_dtype': 'bfloat16', 'bnb_4bit_quant_storage': 'uint8', 'bnb_4bit_quant_type': 'nf4', 'bnb_4bit_use_double_quant': True, 'llm_int8_enable_fp32_cpu_offload': False, 'llm_int8_has_fp16_weight': False, 'llm_int8_skip_modules': [], 'llm_int8_threshold': 6.0}
INFO 06-06 09:53:48 [core.py:65] Initializing a V1 LLM engine (v0.9.0.1) with config: model='unsloth/Phi-3-mini-4k-instruct-v0-bnb-4bit', speculative_config=None, tokenizer='unsloth/Phi-3-mini-4k-instruct-v0-bnb-4bit', skip_tokenizer_init=False, tokenizer_mode=auto, revision=None, override_neuron_config={}, tokenizer_revision=None, trust_remote_code=False, dtype=torch.bfloat16, max_seq_len=1248, download_dir=None, load_format=LoadFormat.BITSANDBYTES, tensor_parallel_size=1, pipeline_parallel_size=1, disable_custom_all_reduce=False, quantization=bitsandbytes, enforce_eager=False, kv_cache_dtype=auto,  device_config=cuda, decoding_config=DecodingConfig(backend='auto', disable_fallback=False, disable_any_whitespace=False, disable_additional_properties=False, reasoning_backend=''), observability_config=ObservabilityConfig(show_hidden_metrics_for_version=None, otlp_traces_endpoint=None, collect_detailed_traces=None), seed=0, served_model_name=unsloth/Phi-3-mini-4k-instruct-v0-bnb-4bit, num_scheduler_steps=1, multi_step_stream_outputs=True, enable_prefix_caching=False, chunked_prefill_enabled=True, use_async_output_proc=True, pooler_config=None, compilation_config={"level": 3, "backend": "inductor", "custom_ops": ["none"], "splitting_ops": ["vllm.unified_attention", "vllm.unified_attention_with_output"], "compile_sizes": [], "inductor_compile_config": {"debug": false, "dce": true, "coordinate_descent_tuning": true, "trace.enabled": false, "trace.graph_diagram": false, "triton.cudagraphs": true, "compile_threads": 48, "max_autotune": false, "disable_progress": false, "verbose_progress": true, "enable_auto_functionalized_v2": false}, "use_cudagraph": true, "cudagraph_num_of_warmups": 1, "cudagraph_capture_sizes": [512, 504, 496, 488, 480, 472, 464, 456, 448, 440, 432, 424, 416, 408, 400, 392, 384, 376, 368, 360, 352, 344, 336, 328, 320, 312, 304, 296, 288, 280, 272, 264, 256, 248, 240, 232, 224, 216, 208, 200, 192, 184, 176, 168, 160, 152, 144, 136, 128, 120, 112, 104, 96, 88, 80, 72, 64, 56, 48, 40, 32, 24, 16, 8, 4, 2, 1], "max_capture_size": 512}
WARNING 06-06 09:53:48 [utils.py:2671] Methods determine_num_available_blocks,device_config,get_cache_block_size_bytes,initialize_cache not implemented in <vllm.v1.worker.gpu_worker.Worker object at 0x7f94d165a350>
INFO 06-06 09:53:49 [parallel_state.py:1064] rank 0 in world size 1 is assigned as DP rank 0, PP rank 0, TP rank 0, EP rank 0
WARNING 06-06 09:53:49 [topk_topp_sampler.py:58] FlashInfer is not available. Falling back to the PyTorch-native implementation of top-p & top-k sampling. For the best performance, please install FlashInfer.
INFO 06-06 09:53:49 [gpu_model_runner.py:1531] Starting to load model unsloth/Phi-3-mini-4k-instruct-v0-bnb-4bit...
INFO 06-06 09:53:49 [cuda.py:217] Using Flash Attention backend on V1 engine.
INFO 06-06 09:53:49 [backends.py:35] Using InductorAdaptor
INFO 06-06 09:53:49 [bitsandbytes_loader.py:454] Loading weights with BitsAndBytes quantization. May take a while ...
INFO 06-06 09:53:49 [weight_utils.py:291] Using model weights format ['*.safetensors']
INFO 06-06 09:53:50 [weight_utils.py:344] No model.safetensors.index.json found in remote.
Loading safetensors checkpoint shards:   0% Completed | 0/1 [00:00<?, ?it/s]
Loading safetensors checkpoint shards: 100% Completed | 1/1 [00:00<00:00,  1.05it/s]
Loading safetensors checkpoint shards: 100% Completed | 1/1 [00:00<00:00,  1.05it/s]

Loading safetensors checkpoint shards:   0% Completed | 0/1 [00:00<?, ?it/s]
Loading safetensors checkpoint shards: 100% Completed | 1/1 [00:01<00:00,  1.07s/it]
Loading safetensors checkpoint shards: 100% Completed | 1/1 [00:01<00:00,  1.07s/it]

INFO 06-06 09:53:52 [punica_selector.py:18] Using PunicaWrapperGPU.
INFO 06-06 09:53:52 [gpu_model_runner.py:1549] Model loading took 2.3478 GiB and 3.170125 seconds
INFO 06-06 09:54:17 [backends.py:459] Using cache directory: /u/scandussio/.cache/vllm/torch_compile_cache/4b45dec40a/rank_0_0 for vLLM's torch.compile
INFO 06-06 09:54:17 [backends.py:469] Dynamo bytecode transform time: 24.85 s
INFO 06-06 09:54:32 [backends.py:132] Directly load the compiled graph(s) for shape None from the cache, took 12.714 s
INFO 06-06 09:54:36 [monitor.py:33] torch.compile takes 24.85 s in total
INFO 06-06 09:54:44 [kv_cache_utils.py:637] GPU KV cache size: 44,272 tokens
INFO 06-06 09:54:44 [kv_cache_utils.py:640] Maximum concurrency for 1,248 tokens per request: 35.47x
INFO 06-06 09:56:00 [gpu_model_runner.py:1933] Graph capturing finished in 76 secs, took 1.35 GiB
INFO 06-06 09:56:01 [core.py:167] init engine (profile, create kv cache, warmup model) took 128.50 seconds
Unsloth: Just some info: will skip parsing ['q_norm', 'k_norm', 'post_feedforward_layernorm', 'pre_feedforward_layernorm']
[rank0]: Traceback (most recent call last):
[rank0]:   File "/share/ai-lab/scandussio/rebus-grpo/grpo-phi.py", line 20, in <module>
[rank0]:     model, tokenizer = FastLanguageModel.from_pretrained(
[rank0]:                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/u/scandussio/.conda/envs/vllm_env/lib/python3.11/site-packages/unsloth/models/loader.py", line 376, in from_pretrained
[rank0]:     model, tokenizer = dispatch_model.from_pretrained(
[rank0]:                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/u/scandussio/.conda/envs/vllm_env/lib/python3.11/site-packages/unsloth/models/mistral.py", line 400, in from_pretrained
[rank0]:     return FastLlamaModel.from_pretrained(
[rank0]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/u/scandussio/.conda/envs/vllm_env/lib/python3.11/site-packages/unsloth/models/llama.py", line 1830, in from_pretrained
[rank0]:     model = convert_vllm_to_huggingface(quant_state_dict, model_config, dtype, bnb_config)
[rank0]:             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/u/scandussio/.conda/envs/vllm_env/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 116, in decorate_context
[rank0]:     return func(*args, **kwargs)
[rank0]:            ^^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/u/scandussio/.conda/envs/vllm_env/lib/python3.11/site-packages/unsloth_zoo/vllm_utils.py", line 665, in convert_vllm_to_huggingface
[rank0]:     new_model = create_empty_causal_lm(config, dtype)
[rank0]:                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/u/scandussio/.conda/envs/vllm_env/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 116, in decorate_context
[rank0]:     return func(*args, **kwargs)
[rank0]:            ^^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/u/scandussio/.conda/envs/vllm_env/lib/python3.11/site-packages/unsloth_zoo/vllm_utils.py", line 651, in create_empty_causal_lm
[rank0]:     new_model = AutoModelForCausalLM.from_config(
[rank0]:                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/u/scandussio/.conda/envs/vllm_env/lib/python3.11/site-packages/transformers/models/auto/auto_factory.py", line 440, in from_config
[rank0]:     return model_class._from_config(config, **kwargs)
[rank0]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/u/scandussio/.conda/envs/vllm_env/lib/python3.11/site-packages/transformers/modeling_utils.py", line 309, in _wrapper
[rank0]:     return func(*args, **kwargs)
[rank0]:            ^^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/u/scandussio/.conda/envs/vllm_env/lib/python3.11/site-packages/transformers/modeling_utils.py", line 2077, in _from_config
[rank0]:     model = cls(config, **kwargs)
[rank0]:             ^^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/u/scandussio/.conda/envs/vllm_env/lib/python3.11/site-packages/transformers/models/mistral/modeling_mistral.py", line 620, in __init__
[rank0]:     self.model = MistralModel(config)
[rank0]:                  ^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/u/scandussio/.conda/envs/vllm_env/lib/python3.11/site-packages/transformers/models/mistral/modeling_mistral.py", line 342, in __init__
[rank0]:     [MistralDecoderLayer(config, layer_idx) for layer_idx in range(config.num_hidden_layers)]
[rank0]:   File "/u/scandussio/.conda/envs/vllm_env/lib/python3.11/site-packages/transformers/models/mistral/modeling_mistral.py", line 342, in <listcomp>
[rank0]:     [MistralDecoderLayer(config, layer_idx) for layer_idx in range(config.num_hidden_layers)]
[rank0]:      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/u/scandussio/.conda/envs/vllm_env/lib/python3.11/site-packages/transformers/models/mistral/modeling_mistral.py", line 223, in __init__
[rank0]:     self.self_attn = MistralAttention(config=config, layer_idx=layer_idx)
[rank0]:                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/u/scandussio/.conda/envs/vllm_env/lib/python3.11/site-packages/transformers/models/mistral/modeling_mistral.py", line 139, in __init__
[rank0]:     self.scaling = self.head_dim**-0.5
[rank0]:                    ~~~~~~~~~~~~~^^~~~~
[rank0]: ZeroDivisionError: 0.0 cannot be raised to a negative power
[rank0]:[W606 09:56:06.369024798 ProcessGroupNCCL.cpp:1476] Warning: WARNING: destroy_process_group() was not called before program exit, which can leak resources. For more info, please see https://pytorch.org/docs/stable/distributed.html#shutdown (function operator())
Traceback (most recent call last):
  File "/u/scandussio/.conda/envs/vllm_env/bin/accelerate", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/u/scandussio/.conda/envs/vllm_env/lib/python3.11/site-packages/accelerate/commands/accelerate_cli.py", line 50, in main
    args.func(args)
  File "/u/scandussio/.conda/envs/vllm_env/lib/python3.11/site-packages/accelerate/commands/launch.py", line 1198, in launch_command
    simple_launcher(args)
  File "/u/scandussio/.conda/envs/vllm_env/lib/python3.11/site-packages/accelerate/commands/launch.py", line 785, in simple_launcher
    raise subprocess.CalledProcessError(returncode=process.returncode, cmd=cmd)
subprocess.CalledProcessError: Command '['/u/scandussio/.conda/envs/vllm_env/bin/python3.11', 'grpo-phi.py']' returned non-zero exit status 1.
DONE!
